src/main/java/pl/wedding/management/service/ExceptionGeneral.java

```diff
@@ -0,0 +1,11 @@
+package pl.wedding.management.service;
+
+public class ExceptionGeneral extends Exception {
```

Comment:
For lambdas, for functional programming it would be better if it extends a RuntimeException

Reply:
zastanawiałem się czy nie zrobić kilku klas właśnie z takimi rozszerzeniami (np IllegalArgExc, NPE, i ten RuntimeException)

Reply:
W ramach tego zadania stwórz również klasę, która obsłuży ten wyjątek i zwróci jakąś odpowiedź z naszego REST'owego API. Dowiedz się czym jest ControllerAdvice. Błąd może zostać rzucony z różnych przyczyn co znaczy, że w zależności od źródła problemu powinniśmy odpowiadać różnymi kodami HTTP. Zastanów się jak to obsłużyć. W razie problemów - pytaj śmiało. :)

Reply:
> zastanawiałem się czy nie zrobić kilku klas właśnie z takimi rozszerzeniami (np IllegalArgExc, NPE, i ten RuntimeException)

IMHO, na to przyjdzie jeszcze czas.

Reply:
ok, czy do tego @ControllerAdvice stworzyć osobną klasę, aby móc jej używać tylko w controllerach? Czy do tej klasy(ExceptionGeneral) dodać tę adnotację? Ja uważam, żeby stworzyć osobną klasę do @ControllerAdvice, (rozdzielić odpowiedzialność) i osobną klasę do błędów w domain.

Reply:
Nie używa się klasy oznaczonej adnotacją ControllerAdvice jako tako. Ona sama łapie wyjątki i zwracana odpowiednie kody. Sam wpadłeś na ten właściwy sposób - osobna klasa.